### ISSUE TITLE
WizardLM inference error: ggml-metal.m:773: false && "not implemented"

### ISSUE THREAD

**Issue by clarkmcc** [rustformers/llm#383]
I'm getting the following error when trying to run the [WizardLM-13B Q8](https://huggingface.co/localmodels/WizardLM-13B-v1.1-ggml/resolve/main/wizardlm-13b-v1.1.ggmlv3.q8_0.bin) model. I'm running this library in a tauri app, let me know if you need any more details or testing from me. I'm running Apple M1 Max (64GB).

```
ggml-sys-8f6d0ee10141006f/out/ggml-metal.m:773: false && "not implemented"
ggml_metal_graph_compute_block_invoke: encoding node 186, op = RMS_NORM
```

**Comment by LLukas22** [comment]
Could you try another quantization format? Maybe q5_1 or one of the K-quantz?

**Comment by clarkmcc** [comment]
Yeah, so [vicuna 7b 2b k-quant](https://huggingface.co/localmodels/Vicuna-7B-v1.3-ggml/resolve/main/vicuna-7b-v1.3.ggmlv3.q2_K.bin) works. Also, if it wasn't clear from the error message, this is actually a panic from the library, (probably an `unimplemented!()` call somewhere?).

**Comment by LLukas22** [comment]
The error seams to be caused by [this](https://github.com/ggerganov/llama.cpp/blob/b7647436ccc80970b44a270f70f4f2ea139054d1/ggml-metal.m#L758-L774) codeblock in the ggml metal shader implementation. We probably have to pull the latest changes into our repo or we have to check if our way of embedding the shader code into the `ggml-metal.m` file creates some issues. Probably something for @philpax. 

@clarkmcc Could you check if these models run on the `llama.cpp` main branch? We use it as our current `ggml` source and if the error is in the shader code we have to create an issue there.

**Comment by clarkmcc** [comment]
@LLukas22 Running the following command seems to work just fine for me
```shell
./main -m ~/models/wizardlm-13b-v1.1.ggmlv3.q4_K_M.bin -n 128 -ngl 1 -p "The meaning of life is "
```

**Comment by philpax** [comment]
We probably need to implement our implementation of the LLaMA model. Not sure if I'll be able to get around to that soon.

**Comment by LLukas22** [comment]
According to https://github.com/ggerganov/llama.cpp/issues/2508 some quantizatio90ns are simply not implemented in metal.
